### PR TITLE
FOGL-4599 Plugin remove unit tests coverage added

### PR DESCRIPTION
**Pull request by ashish-jabble**
Signed-off-by: ashish-jabble <ashish@dianomic.com>

### CODE REVIEW

File: tests/unit/python/fledge/services/core/api/plugins/test_remove.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    async def test_package_when_not_in_use(self, client):
+        async def async_mock(return_value):
+            return return_value
+        _type = "south"
+        name = 'http_south'
+        pkg_name = "fledge-south-http-south"
+        payload = {"return": ["status"], "where": {"column": "action", "condition": "=", "value": "purge",
+                                                   "and": {"column": "name", "condition": "=", "value": pkg_name}}}
+        select_row_resp = {'count': 1, 'rows': [{
+            "id": "c5648940-31ec-4f78-a7a5-b1707e8fe578",
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/plugins/test_remove.py:196]: this for last failure call check?  ID does not matter as patched here, but should be unique on new request insert

File: tests/unit/python/fledge/services/core/api/plugins/test_remove.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        }]}
+        insert = {"response": "inserted", "rows_affected": 1}
+        insert_row = {'count': 1, 'rows': [{
+                "id": "c5648940-31ec-4f78-a7a5-b1707e8fe578",
+                "name": pkg_name,
+                "action": "purge",
+                "status": -1,
+                "log_file_uri": ""
+            }]}
+        delete = {"response": "deleted", "rows_affected": 1}
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/plugins/test_remove.py:210]: this is to verify that select_row_resp that last failed entry gets removed on new request?

**Review by praveen-garg — APPROVED** [review]
We need to ensure, we update `available` cache when process completes with exit code 0.

File: tests/unit/python/fledge/services/core/api/plugins/test_remove.py (hunk 2 above)
Comment (ashish-jabble) [reply tests/unit/python/fledge/services/core/api/plugins/test_remove.py:210]: It is for the deletion of entry from database only when status code is Not in progress mode (i.e -1)

File: tests/unit/python/fledge/services/core/api/plugins/test_remove.py (hunk 1 above)
Comment (ashish-jabble) [reply tests/unit/python/fledge/services/core/api/plugins/test_remove.py:196]: This is the return value (And expected one). For return values as close as with real storage result.

File: tests/unit/python/fledge/services/core/api/plugins/test_remove.py (hunk 2 above)
Comment (praveen-garg) [reply tests/unit/python/fledge/services/core/api/plugins/test_remove.py:210]: So We expect  delete to be called with what we have in `select_row_resp` (failed or success) and insert with new ID for new request if plugin is discoverable. right?

**Pull request merged**